modules/datalayer/matillion.tf

```
@@ -28,13 +28,14 @@ resource "aws_acm_certificate" "matillion" {
 
 module "matillion" {
   source                    = "github.com/schubergphilis/terraform-aws-mcaf-matillion"
+  alb_certificate_arn       = aws_acm_certificate.matillion.arn
+  alb_http_in_cidrs         = ["195.66.90.59/32", "195.66.90.65/32", "89.255.202.50/32"]
   availability_zones        = module.vpc.azs
-  subnet_ids                = module.vpc.public_subnets
-  key_name                  = local.keypair_name
   create_alb                = true
   create_global_accelerator = true
-  alb_certificate_arn       = aws_acm_certificate.matillion.arn
-  alb_http_in_cidrs         = ["195.66.90.59/32"]
-  vpc_id                    = module.vpc.vpc_id
+  instance_type             = var.mat_instance_type
```

It's better to set variables directly whenever possible like so:

```suggestion
  instance_type             = "m5.large"
```

As you're setting the value that is the variable default, IMO would be better to remove the abstraction and just specify the desired value.

We agreed that everything should be in the datalayer module, so if we need to make prod and non prod differ it would be better to do something like this instead of adding a variable to the datalayer module like so:

```suggestion
  instance_type             = var.environment == "prod" ? "<prod-instance-type" : "m5.large"
```

Of course happy to chat about changing this "rule" if preferences have changed! Whatever makes it easier to understand what is going on..

---

as discussed by phone